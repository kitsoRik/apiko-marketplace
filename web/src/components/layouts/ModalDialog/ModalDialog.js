import React from 'react';

import "./ModalDialog.scss";
import Form from '../Form';

const ModalDialog = ({ className, opened, onClosed, children, ...props }) => {
    return (
        <div style={{ display: opened ? "flex" : "none", }}
            className={`modal-dialog`}
            onClick={onClosed}
            {...props}>
            <Form className={`modal-dialog-form ${className ?? ""}`} onClick={e => e.stopPropagation()}>
                {children}
                <button className="modal-dialog-form-close-button" onClick={onClosed}>X</button>
            </Form>
        </div>
    )
};

export default ModalDialog;