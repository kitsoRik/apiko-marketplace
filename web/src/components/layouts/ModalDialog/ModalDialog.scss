.modal-dialog {
    width: 100%;
    height: 100%;

    position: fixed;
    top: 0;
    left: 0;

    display: flex;
    justify-content: center;
    align-items: center;

    background-color: #0005;

    &-form {
        padding: 17px 80px 55px;
        position: relative;
        &-close-button {
            position: absolute;

            top: 0;
            right: 0;
        }
    }
}
