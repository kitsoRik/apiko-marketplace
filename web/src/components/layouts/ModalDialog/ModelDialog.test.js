import React from "react";
import { create } from "react-test-renderer";
import ModalDialog from "./ModalDialog";
import { mount } from "enzyme";

describe("ModalDialog", () => {
    it("should be rendered", () => {
        const component = create(<ModalDialog></ModalDialog>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with children", () => {
        const component = create(<ModalDialog><div><span></span></div></ModalDialog>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render opened", () => {
        const component = create(
            <ModalDialog onClosed={() => { }} opened={true}>
                <div>
                    <span></span>
                </div>
            </ModalDialog>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render closed", () => {
        const component = create(
            <ModalDialog onClosed={() => { }} opened={false}>
                <div>
                    <span></span>
                </div>
            </ModalDialog>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with style", () => {
        const component = create(
            <ModalDialog style={{}}>
                <div>
                    <span></span>
                </div>
            </ModalDialog>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render with classname", () => {
        const component = create(
            <ModalDialog className="custom-class-name">
                <div>
                    <span></span>
                </div>
            </ModalDialog>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should render the loading with children", () => {
        const onClosed = jest.fn();
        const component = mount(<ModalDialog onClosed={onClosed} opened={true}><div><span></span></div></ModalDialog>);

        expect(onClosed).toBeCalledTimes(0);
        component.find("div").first().simulate("click");
        expect(onClosed).toBeCalled();
    });
});
