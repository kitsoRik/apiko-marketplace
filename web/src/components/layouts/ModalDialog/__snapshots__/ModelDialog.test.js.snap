// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialog should be rendered 1`] = `
<div
  className="modal-dialog"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="form modal-dialog-form "
    onClick={[Function]}
  />
</div>
`;

exports[`ModalDialog should render closed 1`] = `
<div
  className="modal-dialog"
  onClick={[Function]}
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="form modal-dialog-form "
    onClick={[Function]}
  >
    <div>
      <span />
    </div>
  </div>
</div>
`;

exports[`ModalDialog should render opened 1`] = `
<div
  className="modal-dialog"
  onClick={[Function]}
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="form modal-dialog-form "
    onClick={[Function]}
  >
    <div>
      <span />
    </div>
  </div>
</div>
`;

exports[`ModalDialog should render with children 1`] = `
<div
  className="modal-dialog"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="form modal-dialog-form "
    onClick={[Function]}
  >
    <div>
      <span />
    </div>
  </div>
</div>
`;

exports[`ModalDialog should render with classname 1`] = `
<div
  className="modal-dialog"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="form modal-dialog-form custom-class-name"
    onClick={[Function]}
  >
    <div>
      <span />
    </div>
  </div>
</div>
`;

exports[`ModalDialog should render with style 1`] = `
<div
  className="modal-dialog"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="form modal-dialog-form "
    onClick={[Function]}
    style={Object {}}
  >
    <div>
      <span />
    </div>
  </div>
</div>
`;
